File: GuiaApe06/Ejercicio5.cpp
#include <iostream>   // Librería para entrada y salida de datos
#include <vector>     // Librería para usar vectores dinámicos

using namespace std;

int main() {

    // Vector para guardar nombres
    vector<string> nombres;

    // Vector para guardar notas
    vector<double> notas;

    // Variables auxiliares
    int cantidad;
    string nombre;
    double nota;

    // Pedir cantidad de estudiantes
    cout << "Cantidad de estudiantes: ";
    cin >> cantidad;

    // Limpiar buffer
    cin.ignore();

    // Ciclo para registrar estudiantes
    for(int i = 0; i < cantidad; i++) {

        // Pedir nombre del estudiante
        cout << "Nombre: ";

        // Leer nombre completo
        getline(cin, nombre);

        // Pedir nota del estudiante
        cout << "Nota: ";

        // Leer nota
        cin >> nota;

        // Limpiar buffer
        cin.ignore();

        // Guardar nombre en el vector
        nombres.push_back(nombre);

        // Guardar nota en el vector
        notas.push_back(nota);
    }

    // Mostrar título
    cout << "\nREGISTROS\n";

    // Recorrer y mostrar registros
    for(int i = 0; i < nombres.size(); i++) {

        // Mostrar nombre y nota
        cout << nombres[i] << " - " << notas[i] << endl;
    }

    return 0;
}

File: GuiaApe06/Ejercicio5.java
import java.util.ArrayList; // Librería para usar ArrayList
import java.util.Scanner;   // Librería para entrada de datos

public class Ejercicio5{

    public static void main(String[] args) {

        // Crear objeto Scanner para leer datos
        Scanner sc = new Scanner(System.in);

        // ArrayList para guardar nombres
        ArrayList<String> nombres = new ArrayList<>();

        // ArrayList para guardar notas
        ArrayList<Double> notas = new ArrayList<>();

        // Variables auxiliares
        int cantidad;
        String nombre;
        double nota;

        // Pedir cantidad de estudiantes
        System.out.print("Cantidad de estudiantes: ");
        cantidad = sc.nextInt();

        // Limpiar buffer
        sc.nextLine();

        // Ciclo para registrar estudiantes
        for(int i = 0; i < cantidad; i++) {

            // Pedir nombre del estudiante
            System.out.print("Nombre: ");

            // Leer nombre
            nombre = sc.nextLine();

            // Pedir nota
            System.out.print("Nota: ");

            // Leer nota
            nota = sc.nextDouble();

            // Limpiar buffer
            sc.nextLine();

            // Guardar nombre en el ArrayList
            nombres.add(nombre);

            // Guardar nota en el ArrayList
            notas.add(nota);
        }

        // Mostrar título
        System.out.println("\nREGISTROS");

        // Recorrer y mostrar registros
        for(int i = 0; i < nombres.size(); i++) {

            // Mostrar nombre y nota
            System.out.println(nombres.get(i) + " - " + notas.get(i));
        }

        // Cerrar Scanner
        sc.close();
    }
}
